### PR TITLE
chore: remove headless radio and use react-aria component instead

### LINKED ISSUE #27677: [Bug]: Checkbox when being used in canvas is getting focus even on click
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When we used our new WDS checkbox, the checkbox is getting the focus on click as well. The focus should only come on when using keyboard. 
Also, the checkbox outside canvas works fine.

https://github.com/appsmithorg/appsmith/assets/6636360/72f49e30-4cac-49e2-be16-7824b3a5be74

### Steps To Reproduce

Drop the newly made checkbox widget v2, and drop it on canvas. and click on it multiple times. You will see the focus styles coming up.

### Public Sample App

_No response_

### Environment

Production

### Issue video log

_No response_

### Version

Latest

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description

- Remove headless radio and use react-aria component instead
- Create ErrorMessage component

Fixes #27677 

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9567388261>
> Commit: 62d1153caa8bf03d827f88593c9dfaf3121091ee
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9567388261&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Anvil`

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
  - Added `ErrorMessage` component for displaying error messages with specific styling.
  - Introduced `isDisabled` prop for `Label` and `ToggleGroup` components.
  - Updated `RadioGroup` to accept an items array for easier configuration.

- **Bug Fixes**
  - Improved conditional rendering in `Label` component to prevent issues when `text` and `contextualHelp` are both falsy.

- **Refactor**
  - Removed `Radio` component export from design system.
  - Restructured import statements and prop declarations for `Checkbox` and `RadioGroup`.

- **Style**
  - Updated styles for `RadioGroup` and `ToggleGroup` components for better state handling (disabled, hovered, selected).

- **Documentation**
  - Updated Storybook stories for `RadioGroup` to reflect changes in component usage.

- **Tests**
  - Adjusted `RadioGroup.test.tsx` to test new items array prop.

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/ToggleGroup/src/ToggleGroup.tsx
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/ToggleGroup/src/ToggleGroup.tsx:1]: QQ: didn't we decide to remove toggle group ( I remember a thread on slack you asking about it )

File: app/client/packages/design-system/widgets/src/components/RadioGroup/src/RadioGroup.tsx [hunk 1]
```diff
@@ -1,35 +1,96 @@
...
-import { ContextualHelp } from "../../ContextualHelp";
+interface RadioGroupItemProps {
+  value: string;
+  label?: string;
+  isSelected?: boolean;
+  isDisabled?: boolean;
+  index?: number;
+}
 
 export interface RadioGroupProps extends HeadlessRadioGroupProps {
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/RadioGroup/src/RadioGroup.tsx]: Fixed

File: app/client/packages/design-system/widgets/src/components/RadioGroup/src/styles.module.css [hunk 2]
```diff
@@ -11,11 +32,10 @@
       var(--radio-border-color) inset;
 
     /**
-      Checkbox icon are positioned absolutely because we need to align the elements along the baseline
+      Radio icon are positioned absolutely because we need to align the elements along the baseline
```
[the remark this answers is not in the document]
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/RadioGroup/src/styles.module.css]: Fixed

File: app/client/packages/design-system/widgets/src/components/ToggleGroup/src/ToggleGroup.tsx
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/ToggleGroup/src/ToggleGroup.tsx:1]: Nope, Taras said that we need it.

**Review by jsartisan — APPROVED** [review]

**Pull request merged**